Index: ks-common-api/src/main/java/org/kuali/student/r2/common/util/constants/CourseRegistrationServiceConstants.java
===================================================================
--- ks-common-api/src/main/java/org/kuali/student/r2/common/util/constants/CourseRegistrationServiceConstants.java	(revision 60873)
+++ ks-common-api/src/main/java/org/kuali/student/r2/common/util/constants/CourseRegistrationServiceConstants.java	(working copy)
@@ -21,5 +21,8 @@
 public class CourseRegistrationServiceConstants {
     
     public static final String NAMESPACE = CommonServiceConstants.REF_OBJECT_URI_GLOBAL_PREFIX + "courseRegistrationService";
+//    public static final String SERVICE_NAME_LOCAL_PART = CourseRegistrationService.class.getSimpleName();
+    // TODO: move this file to enrollment api so we can reference the CR service like above
+    public static final String SERVICE_NAME_LOCAL_PART = "CourseRegistrationService";
 
 }
Index: ks-enroll-api/src/main/java/org/kuali/student/enrollment/academicrecord/dto/LoadInfo.java
===================================================================
--- ks-enroll-api/src/main/java/org/kuali/student/enrollment/academicrecord/dto/LoadInfo.java	(revision 60873)
+++ ks-enroll-api/src/main/java/org/kuali/student/enrollment/academicrecord/dto/LoadInfo.java	(working copy)
@@ -11,6 +11,7 @@
 import javax.xml.bind.annotation.XmlType;
 import java.io.Serializable;
 import java.util.List;
+import org.kuali.rice.core.api.util.type.KualiDecimal;
 
 /**
  * @Version 2.0
@@ -24,7 +25,7 @@
 public class LoadInfo extends IdNamelessEntityInfo implements Load, Serializable {
     private static final long serialVersionUID = 1L;
     @XmlElement
-    private String totalCredits;
+    private KualiDecimal totalCredits;
     @XmlElement
     private String loadLevelTypeKey;
     @XmlAnyElement
@@ -43,11 +44,11 @@
     }
 
     @Override
-    public String getTotalCredits() {
+    public KualiDecimal getTotalCredits() {
         return totalCredits;
     }
 
-    public void setTotalCredits(String totalCredits) {
+    public void setTotalCredits(KualiDecimal totalCredits) {
         this.totalCredits = totalCredits;
     }
 
Index: ks-enroll-api/src/main/java/org/kuali/student/enrollment/academicrecord/infc/Load.java
===================================================================
--- ks-enroll-api/src/main/java/org/kuali/student/enrollment/academicrecord/infc/Load.java	(revision 60873)
+++ ks-enroll-api/src/main/java/org/kuali/student/enrollment/academicrecord/infc/Load.java	(working copy)
@@ -1,5 +1,6 @@
 package org.kuali.student.enrollment.academicrecord.infc;
 
+import org.kuali.rice.core.api.util.type.KualiDecimal;
 import org.kuali.student.r2.common.infc.IdNamelessEntity;
 
 /**
@@ -14,7 +15,7 @@
      * @name Total Credits
      * @required
      */
-    public String getTotalCredits();
+    public KualiDecimal getTotalCredits();
 
     /**
      * Code indicating the load level, for example, Full Time, Half Time
Index: ks-enroll-api/src/main/java/org/kuali/student/enrollment/courseregistration/dto/CourseRegistrationInfo.java
===================================================================
--- ks-enroll-api/src/main/java/org/kuali/student/enrollment/courseregistration/dto/CourseRegistrationInfo.java	(revision 60873)
+++ ks-enroll-api/src/main/java/org/kuali/student/enrollment/courseregistration/dto/CourseRegistrationInfo.java	(working copy)
@@ -31,9 +31,18 @@
 
 @XmlAccessorType(XmlAccessType.FIELD)
 @XmlType(name = "CourseRegistrationInfo", propOrder = {
-        "id", "typeKey", "stateKey", 
-        "studentId", "courseOfferingId", "credits", "gradingOptionId", 
-        "effectiveDate", "expirationDate", "meta", "attributes", 
+        "id", 
+        "typeKey", 
+        "stateKey", 
+        "studentId", 
+        "courseOfferingId", 
+        "registrationGroupId", 
+        "credits", 
+        "gradingOptionId", 
+        "effectiveDate", 
+        "expirationDate", 
+        "meta", 
+        "attributes", 
         "_futureElements"})
 
 public class CourseRegistrationInfo 
@@ -49,6 +58,9 @@
     private String courseOfferingId;
 
     @XmlElement
+    private String registrationGroupId;
+    
+    @XmlElement
     private String credits;
 
     @XmlElement
@@ -76,6 +88,7 @@
         if (courseRegistration != null) {
             this.studentId = courseRegistration.getStudentId();
             this.courseOfferingId = courseRegistration.getCourseOfferingId();
+            this.registrationGroupId = courseRegistration.getRegistrationGroupId();
             this.credits = courseRegistration.getCredits();
             this.gradingOptionId = courseRegistration.getGradingOptionId();
          }
@@ -100,6 +113,15 @@
     }
 
     @Override
+    public String getRegistrationGroupId() {
+        return registrationGroupId;
+    }
+
+    public void setRegistrationGroupId(String registrationGroupId) {
+        this.registrationGroupId = registrationGroupId;
+    } 
+
+    @Override
     public String getCredits() {
         return credits;
     }
Index: ks-enroll-api/src/main/java/org/kuali/student/enrollment/courseregistration/dto/CreditLoadInfo.java
===================================================================
--- ks-enroll-api/src/main/java/org/kuali/student/enrollment/courseregistration/dto/CreditLoadInfo.java	(revision 60873)
+++ ks-enroll-api/src/main/java/org/kuali/student/enrollment/courseregistration/dto/CreditLoadInfo.java	(working copy)
@@ -24,6 +24,7 @@
 import javax.xml.bind.annotation.XmlAnyElement;
 import javax.xml.bind.annotation.XmlElement;
 import javax.xml.bind.annotation.XmlType;
+import org.kuali.rice.core.api.util.type.KualiDecimal;
 
 import org.kuali.student.enrollment.courseregistration.infc.CreditLoad;
 import org.w3c.dom.Element;
@@ -42,13 +43,13 @@
     private String studentId;
 
     @XmlElement
-    private String creditLoad;
+    private KualiDecimal creditLoad;
 
     @XmlElement
-    private String creditLimit;
+    private KualiDecimal creditLimit;
 
     @XmlElement
-    private String additionalCredits;
+    private KualiDecimal additionalCredits;
 
     @XmlAnyElement
     private List<Element> _futureElements;
@@ -85,29 +86,29 @@
     }
 
     @Override
-    public String getCreditLoad() {
+    public KualiDecimal getCreditLoad() {
         return creditLoad;
     }
 
-    public void setCreditLoad(String creditLoad) {
+    public void setCreditLoad(KualiDecimal creditLoad) {
         this.creditLoad = creditLoad;
     }
 
     @Override
-    public String getCreditLimit() {
+    public KualiDecimal getCreditLimit() {
         return creditLimit;
     }
 
-    public void setCreditLimit(String creditLimit) {
+    public void setCreditLimit(KualiDecimal creditLimit) {
         this.creditLimit = creditLimit;
     }
 
     @Override
-    public String getAdditionalCredits() {
+    public KualiDecimal getAdditionalCredits() {
         return additionalCredits;
     }
 
-    public void setAdditionalCredits(String credits) {
-        this.additionalCredits = credits;
+    public void setAdditionalCredits(KualiDecimal additionalCredits) {
+        this.additionalCredits = additionalCredits;
     }
 }
Index: ks-enroll-api/src/main/java/org/kuali/student/enrollment/courseregistration/infc/CourseRegistration.java
===================================================================
--- ks-enroll-api/src/main/java/org/kuali/student/enrollment/courseregistration/infc/CourseRegistration.java	(revision 60873)
+++ ks-enroll-api/src/main/java/org/kuali/student/enrollment/courseregistration/infc/CourseRegistration.java	(working copy)
@@ -52,6 +52,17 @@
     public String getCourseOfferingId();
 
     /**
+     * The registration group Id for this course registration.
+     * 
+     * @name Registration Group Id
+     * @readOnly
+     * @required
+     * @impl Lpr.LuiId
+     */
+    public String getRegistrationGroupId();
+    
+    
+    /**
      * The number of credits for which the student
      * registered. Typically, this number should be in the range
      * defined at the CourseOffering or RegistrationGroup.
Index: ks-enroll-api/src/main/java/org/kuali/student/enrollment/courseregistration/infc/CreditLoad.java
===================================================================
--- ks-enroll-api/src/main/java/org/kuali/student/enrollment/courseregistration/infc/CreditLoad.java	(revision 60873)
+++ ks-enroll-api/src/main/java/org/kuali/student/enrollment/courseregistration/infc/CreditLoad.java	(working copy)
@@ -16,6 +16,8 @@
 
 package org.kuali.student.enrollment.courseregistration.infc;
 
+import org.kuali.rice.core.api.util.type.KualiDecimal;
+
 /**
  * This summarizes the credit load for a student.
  */
@@ -34,14 +36,14 @@
      * 
      * @name Credit Load
      */
-    public String getCreditLoad();
+    public KualiDecimal getCreditLoad();
 
     /**
-     * The maxmimum credit limitfor the student in the current term.
+     * The maxmimum credit limit for the student in the current term.
      * 
      * @name Credit Limit
      */
-    public String getCreditLimit();
+    public KualiDecimal getCreditLimit();
 
     /**
      * The additional credits from the RegistrationRequest. For drops,
@@ -49,5 +51,5 @@
      * 
      * @name Additional Credits
      */
-    public String getAdditionalCredits();
+    public KualiDecimal getAdditionalCredits();
 }
